src/components/pipeline-list/index.js

```diff
@@ -29,7 +30,7 @@ export const PipelineList = ({
         theme={theme}
         width={null}
         onChanged={onUpdateActivePipeline}
-        defaultText={pipeline.name[pipeline.active]}>
+        defaultText={pipeline.name[pipeline.active] || ''}>
```

Comment:
While this seems to solve the issue, I don't think just an empty label works here at all, as it appears broken and functions weirdly when you click on it.

My suggestion is when there's a single pipeline is to always show `Default` or similar, with a corresponding `MenuOption` too for consistency (@GabrielComymQB is probably best to confirm)

I think that is easily achieved in here as a default label (i.e. placeholder) option in this component, rather than needing any further business logic.

Reply:
1. Why do you say it appears broken/functions weirdly? I mean, granted, if you click on it then it will fade the main sidebar bg without displaying any results, but that can't happen anyway unless the results don't load or there aren't any pipelines in the dataset, which Yetu assures us can't happen. But I don't think it looks weird, I think it looks like an empty visual placeholder.
2. Due to the second set of changes, here, you will only ever see the blank pipeline selector for less than half a second on first page-load, while the JSON file is being loaded and parsed. It is not shown when switching pipeline - I've changed it so it still shows the old dataset in the sidebar when you're switching pipeline.
3. I oppose using 'Default' as the placeholder, because if you are running a different pipeline as main, OR if you have a different active pipeline saved in localstorage (e.g. because you just refreshed the page), then on load it will flash 'Default' briefly then switch to your actual pipeline name.

This is how it looks on page-load:
![onload-pipeline-selector](https://user-images.githubusercontent.com/1155816/98371365-f58cd300-2033-11eb-8478-e8c6e4c26153.gif)

Reply:
Ah well here's what I'm seeing when I load a pipeline JSON file with only one pipeline and it stays this way permanently:

![pipeline-selector-blank](https://user-images.githubusercontent.com/45365769/98379925-cb411280-203f-11eb-841b-8894fe19cd74.gif)

So #1 seems to happen at least in this case? Or do the new API changes mean this won't happen in practice (what will it say in that case)?

> I don't think it looks weird, I think it looks like an empty visual placeholder.

It's not really a placeholder if it's empty though surely? A placeholder is meant to fill in gaps like these?

> I oppose using 'Default' as the placeholder, because if you are running a different pipeline as main

Could it just say `Main` then?

If you really think blank is the right thing here, then I'd say also hide the down arrow icon and disable and gray out the field so it can't be interactive?

Reply:
ok do you mean one that does not have any pipelines in the dataset? Because if it has only one it should look like this (when open):
![image](https://user-images.githubusercontent.com/1155816/98385954-84efb180-2047-11eb-893a-e05d62f0d4a3.png)
Whereas I'm able to replicate your case if I remove all pipelines from the dataset.

I changed it to always show whenever data is being loaded asynchronously, only because, as Yetu said on slack, due to recent API changes, it should never be the case that the JSON data doesn't include the pipelines field with at least one pipeline in it, which is why I changed this setting. If we want to support async datasets with 0 pipelines then we should revert this change. Personally, that's what I would recommend, and it's the way I had it set up before, because it is less likely to look 'broken'.

So to summarise:
I misunderstood what you were talking about before - I thought you were just talking about the brief flash on load, where it functions as a placeholder. It seems like you're talking about a zero-pipeline async dataset, which is a case that I was previously was supporting, but was instructed to stop supporting.

I am going to revert this change in the interests of prioritising platform stability across incomplete datasets, over smooth page-loads.